src/controllers/activityController.ts

```diff
@@ -0,0 +1,21 @@
+async function getSwimmingActivity(req: Request, res: Response, next: NextFunction) {
+    try {
+        const activityId = req.query.id;
+        if(activityId === '*'){
+            const activties:swimmingType[] = await swimmingTable.find({});
+            res.json(activties);
+        }
+        else if (activityId) {
```

Comment:
You are missing a logical expression check if the activity exits and if it does carry on the function